enhancements/network/bgp-ovn-kubernetes.md

```diff
@@ -0,0 +1,701 @@
+and thus reducing the overhead and byte size of each packet. This allows for greater throughput.
+
+### Multi-homing, Link Redundancy, Fast Convergence
+
+BGP can use multi-homing with ECMP routing in order to provide layer 3 failover. When a link goes down, BGP can reroute
+via a different path. This functionality can be coupled with BFD in order to provide fast failover.
+
+### User Stories
+
+ * As a user I want to be able to leverage my existing BGP network to dynamically learn routes to pods in my Kubernetes
+   cluster.
+ * As a user, rather than having to maintain routes with NM State manually in each Kubernetes node, as well as being
+   constrained to using local gateway mode for respecting user defined routes; I want to use BGP so that I can dynamically
+   advertise egress routes for the Kubernetes pod traffic in either gateway mode.
+ * As a user where maximum throughput is a priority, I want to reduce packet overhead by not having to encapsulate
+   traffic with Geneve.
+ * As a baremetal or egress IP user, I do not want to have to restrict my nodes to the same layer 2 segment and prefer
+   to use a pure routing implementation to handle advertising virtual IP (VIP) movement across nodes.
+
+### Goals
+
+* To provide a user facing API to allow configuration of iBGP or eBGP peers, along typical BGP configuration including
+  communities, route filtering, etc.
+* Support for advertising Egress IP addresses.
+* To enable BFD to BGP peers.
+* Leveraging BGP to allow for no overlay encapsulation with east/west traffic.
+* ECMP routing support within OVN for BGP learned routes.
+* Support for advertising user-defined networks via BGP as long as there is no subnet overlap over the default VRF.
+* Allowing for VRF-Lite type of VPN where the user maps interfaces on the host to user-defined VRFs/networks and
+advertises VPN routes via BGP sessions over said VRFs.
+
+### Non-Goals
+
+* Support of any other routing protocol.
+* Running separate BGPd instances per VRF network.
+* Providing any type of API or operator to automatically connect two Kubernetes clusters via L3VPN.
+* Replacing the support that MetalLB provides today for advertising service IPs.
+* Support for any other type of BGP speaker other than FRR.
+
+### Future Goals
+
+* Support EVPN configuration and integration with a user’s DC fabric, along with MAC-VRFs and IP-VRFs.
+* Support iBGP with route reflectors.
+* Potentially advertising other IP addresses, including the Kubernetes API VIP across the BGP fabric.
+
+## Proposal
+
+OVN-Kubernetes will leverage other projects that already exist to enable BGP in Linux. FRR will be used as the BGP
+speaker and already has EVPN support for native Linux constructs like Linux bridges, VRF devices, VXLAN tunnels, etc.
+FRR may need some code contributions to allow it to integrate with OVN and Open vSwitch. For FRR configuration, the
```

Comment:
"some"? Is this a trivial amount of work or a major part of the overall BGP-in-OVNK project?

Reply:
I think I copied this from the draft document that included EVPN and did not reword it. For detailed info about the work needed in OVN see: https://issues.redhat.com/browse/FDP-670?focusedId=25089147&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-25089147

Requirements 1 and 2 are applicable to this enhancement. I'll try to reword things in the enhancement to make more sense.